mlflow/langchain/_langchain_autolog.py

```diff
@@ -213,8 +215,21 @@ def __init__(self, *args, **kwargs):
         )
 
     run_id = getattr(self, "run_id", None)
-    if (active_run := mlflow.active_run()) and run_id is None:
-        run_id = active_run.info.run_id
+    active_run = mlflow.active_run()
+    if run_id is None:
+        # only log the tags once
+        extra_tags = get_autologging_config(mlflow.langchain.FLAVOR_NAME, "extra_tags", None)
+        # include run context tags
+        resolved_tags = context_registry.resolve_tags(extra_tags)
+        tags = _resolve_extra_tags(mlflow.langchain.FLAVOR_NAME, resolved_tags)
```

Comment:
Q: Why not moving these lines into `if active_run:` branch ?

Reply:
Because even if active_run is None, we should generate those tags, and tags are used inside MLflowCallbackHandler to create a new run (and logs it). But we only log the tags if active_run is not None.